java/arcs/core/host/SimpleSchedulerProvider.kt

```diff
@@ -0,0 +1,37 @@
+package arcs.core.host
+
+import arcs.core.util.Scheduler
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.Job
+
+/**
+ * Simple implementation of a [SchedulerProvider].
+ */
+class SimpleSchedulerProvider(
+    baseCoroutineContext: CoroutineContext
+) : SchedulerProvider {
+    private val providerParentJob = Job(baseCoroutineContext[Job])
+    private val providerCoroutineContext = baseCoroutineContext + providerParentJob
+    private val schedulers = mutableMapOf<String, Scheduler>()
+
+    @Synchronized
```

Comment:
Mostly optional, since we likely have this in use elsewhere in the codebase, but I'm pretty sure that @Synchronized is jvm-only, or at least not supported for kotlin-native. https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-synchronized/

This seems like a scenario where an atomic reference and a map + Ray's lens stuff could work well.

Reply:
I was considering removing the map completely, and having the contract for this object be that it always creates a new scheduler.

There's one place internally where we call into the scheduleProvider again with an arcId to get an existing scheduler. But at that callsite, we also have access the the actual scheduler itself, so the call isn't really needed.

Are there situations where we might need this functionality?